lib/Basics/Utf8Helper.cpp

```diff
@@ -128,15 +128,16 @@ bool Utf8Helper::setCollatorLanguage(std::string_view lang,
   status = U_ZERO_ERROR;
 
   if (_coll) {
-    ULocDataLocaleType type = ULOC_ACTUAL_LOCALE;
+    ULocDataLocaleType type = ULOC_VALID_LOCALE;
```

Comment:
I _guess_ this change is useful to fix the problem at hand, but I do not overlook if there could be other undesired consequences of this change.
What is the difference between `ULOC_VALID_LOCALE` and `ULOC_ACTUAL_LOCALE`? I don't find the ICU documentation particularly useful.

Reply:
I made following assumption:
1. in `getCollatorLanguage` and in `tokenize` methods we use `ULOC_VALID_LOCALE` to obtain the locale and check some properties in it. 
2. In `setCollatorLanguage` we also obtaining collator and check the language inside it. That is why it looks like that we also need to use it here. 

https://unicode-org.github.io/icu/userguide/locale/#valid-locale
```
Valid Locale
A populated locale is one for which ICU has data, or one in which client code has registered a service. If the requested locale is not populated, then ICU will fallback until it reaches a populated locale. The first populated locale it reaches is the valid locale. The valid locale is reachable from the requested locale via zero or more fallback steps.
```

Reply:
The problem is that I cannot tell from the ICU documentation alone if this change will get us into trouble in case a _different_ locale is used now than before. 
I am under the impression that the locale must not be changed after it was originally selected/configured. Because we store locale-dependent sort orders etc. on disk.

Reply:
OK, I agree. Right now I tried to remove this change and seems that it still works. So my small refactoring in logic has fixed this issue. Will launch Jenkins again